### ISSUE TITLE
Feature request: Animal companion choices

### ISSUE THREAD

**Issue by Crowhazard** [Vek17/WrathMods-TabletopTweaks#133]
I feel like you should be able to choose any animal companions no matter what archetypes and classes since in the tabletop game its up to the gm.

**Comment by Amariithynar** [comment]
Animal companions are restricted to what's available in the local terrain, so the restriction makes a fair amount of sense. Cavaliers and Paladins even moreso are restricted based on their companions being their mounts (in older printings their bond was actually outright called "divine mount" and was JUST a horse).

**Comment by Crowhazard** [comment]
> Animal companions are restricted to what's available in the local terrain, so the restriction makes a fair amount of sense. Cavaliers and Paladins even moreso are restricted based on their companions being their mounts (in older printings their bond was actually outright called "divine mount" and was JUST a horse).

Some archetypes and paladin ofc you are limited but still besides dinosaurs you could make a case for some and imo you can choose a pet that is not of appropriate size for some if he becomes the right size later cause in the tabletop you can change that animal later on which is not the case here.

**Comment by Amariithynar** [comment]
> Some archetypes and paladin ofc you are limited *snip*

No, all are limited. The means of gaining an Animal Companion is, "by performing a ceremony requiring 24 uninterrupted hours of prayer in the environment where the new companion typically lives". Exotic animals captured and held in non-standard terrain areas are possible, of course., but *that* is by GM fiat. I noted that Paladins and Cavaliers are restricted even moreso than the base animal companion feature because of what they use their mounts for.

**Comment by Amariithynar** [comment]
@Carden-Gixoth I already addressed how there are more exotic options that are possibly available with, "Exotic animals captured and held in non-standard terrain areas are possible, of course., but that is by GM fiat." (including my punctuation typo). 

I do wish that Owlcat properly used the "Bonded mounts have an Intelligence of at least 6." aspect, though, either way; having to spend a stat point on int to allow a mount to learn feats with a base int 3 requirement sucks.

**Comment by Carden-Gixoth** [comment]
> I do wish that Owlcat properly used the "Bonded mounts have an Intelligence of at least 6." aspect, though, either way; having to spend a stat point on int to allow a mount to learn feats with a base int 3 requirement sucks.

I didn't know this was an issue. My first playthrough is a Divine Hunter Paladin, which doesn't get a mount. The game says you can choose to get an Animal Companion, or Divine Weapon Bond, (PnP, no you can't) but just shunts you into Divine Weapon Bond; _and_ the DH Weapon Bond is nerfed. Doesn't progress much (if at all) beyond level 8, even though it's supposed to be (Progression wise) a 1:1 with base Paladin.

**Comment by Amariithynar** [comment]
You can see it with Seelah when she hits 6 and gets her mount; it only has an int of 2 and requires a stat level up to be able to qualify for int 3 feats.

As for Divine Hunter, that's because of how the Divine Bond class feature works, split as it is between the two secondary aspects of Animal Companion and Weapon Bond; Divine Hunter accurately cannot get a mount and must get a weapon bond. https://www.d20pfsrd.com/classes/core-classes/paladin/archetypes/paizo-paladin-archetypes/divine-hunter/ Additionally, the benefits it gives are rather minor, but flexible, basically working like the Warpriest's divine weapon enhancements, or the Shaman's Spirit Enhancements (which are really functionally the same). I don't see anything special that Divine Hunter's Weapon Bond should get beyond the default scaling, working just like those other two effects that I mentioned.

**Comment by Carden-Gixoth** [comment]
Ok. The way it's supposed to work is:

- Every 4 levels beyond 5 (to a max of 4 total at level 17) you get more uses of Divine Weapon Bond. 
    - This does not happen in Wrath of the Righteous.
- Every 3 levels beyond level 5, you get an additional +1 (to a max of +6 at level 20), which lets you add more powerful enchantments to your weapon. 
     - Starting out with +1 at level 5, you can add Flaming, Merciful (Not in WotR) Defending, or Keen (Divine Hunters get Distance and Seeking as well, and lose out on Defending). 
     - At level 8, you upgrade to +2 and you can add Flaming Burst, Holy, Axiomatic, or Disruption (if bludgeoning weapon, Divine Hunter does _not_ get Disruption). Alternatively, instead of using just Flaming Burst, you can add both Flaming and Keen, making your weapon (base weapon and all effect) + both Flaming _and_ Keen.
     - At level 11 you upgrade to +3 and you get access to Speed, and can now add three of the +1 effects, one +1 and one +2 effect, or just Speed
     - At level 14 you upgrade to +4, and get access to the final enchantment, Brilliant Energy, which lets your weapons (or ammunition of sling, bow, or crossbow) bypass all nonliving matter, making all attacks effectively Touch Attacks, but also makes them useless against Undead and Constructs, as well as all other nonliving matter. Instead of enchanting with  Brilliant Energy, you can add Flaming, Keen, and Holy/Axiomatic (+1, +1, +2), or (for the Divine Hunter PnP) Seeking, Keen, and Flaming Burst (better threat profile for proccing the additional Fire damage on crit)
     - At level 17, you upgrade to +5, but get no additional enchantments, letting you mix and match any enchantments you have so long as they add up to +5 or less.
     - At level 20, you upgrade to +6, but no additional enchantments still. 
     - It should be noted that you can decide not to add an enchantment to your weapon, making it just a +1-+6 weapon, depending on your Paladin level (Also, this enhancement bonus doesn't stack with the weapon, so if you're using Finnian, you have to be level 14 to make him a +4 bow, though you can forgo that and add any enchantment to him that's not Ghost Touch, or Heartseeker and they work perfectly)
     - it should also be noted, that Divine Hunter Paladins in WotR **_do not scale like this beyond level 8, even though they're supposed to, as the Divine Hunter Divine Weapon Bond (with the exception of gaining Distance and Seeking, and losing out on Disruption and Defending) is _the exact same_ as the default Paladin, including all scaling and additional enchantments.

In Pathfinder: Kingmaker, the Divine Hunter Paladin Divine Weapon Bond scales 1:1 with base Paladin as shown in the attacked screenshots, because this is how the Divine Weapon Bond is _supposed_ to work.
![image](https://user-images.githubusercontent.com/93364891/139541952-cc732a29-effd-4926-81c3-e008dc217362.png)
![image](https://user-images.githubusercontent.com/93364891/139541966-dfc6e941-c694-4a2a-8416-4f9fed2715ce.png)


As for the Mount, I had Seelah take the Divine Weapon Bond (she gets to choose at level 5, not level 6) as the additional damage is amazing. Holy is invaluable during the Crusade, and she can later on make Radiance (the Holy Avenger) be Keen, Flaming Burst, and Axiomatic dealing an additional 2d6 vs Chaotic, doubling Radiance Crit Profile (which doesn't stack with Improved Critical, so a moot point at level 20), and also do +1d6 Fire, +1d10 Fire on Crit (+2d10 Fire cause she has Mythic Improved Critical), and this is the default path for her if you set her to auto level up, so I didn't catch that when leveling her up.

**Comment by Carden-Gixoth** [comment]
Oh, I forgot to reiterate that PnP Divine Hunter _Paladins_ don't get access to an Animal Companion _*at all,*_ Mount or otherwise. Divine Hunter archetype _Hunters,_ however _do_ get an Animal Companion, so I think the issue is that they smashed DH Hunter and DH Paladin together, locked you out of the Animal Companion, even though the tooltip says you can choose one (again, you can't PnP), and didn't scale the DWB properly.
